core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1548,6 +1548,7 @@ public void delete(AlluxioURI path, DeleteContext context)
             for (LockedInodePath childPath : descendants) {
               try {
                 mPermissionChecker.checkPermission(Mode.Bits.WRITE, childPath);
+                mMountTable.checkIsWritableMountPoint(childPath.getUri());
```

Comment:
Instead of introducing a new method, I was thinking more like:

```
if (mMountTable.isMountPoint(childPath.getUri())) {
  mMountTable.checkUnderWritableMountPoint(childPath.getUri());
}
```

This would allow not creating a new method, and also, this performs the (potentially expensive) check for only the mount point paths.

What do you think?

Reply:
I see, this way is much more elegant, thanks!